### ISSUE TITLE
Letters are white rectangle in gui

### ISSUE THREAD

**Issue by lucasParis** [braitsch/ofxDatGui#90]
Hi,

I'm getting this bug where the characters in the gui all become white rectangles.
<img width="966" alt="screen shot 2016-07-26 at 10 36 18" src="https://cloud.githubusercontent.com/assets/7583039/17142144/0c672b7c-531d-11e6-88b0-1847641b35d8.png">

I'm using two windows maybe that could be a problem?

Here is my main gui:

```
void GuiApp::setup(){

    ofBackground(240);
    int columnWidth = (ofGetWidth()-sideBarSize)/3.;
    view = new ofxDatGuiScrollView("ScrollView", 5);
    view->setWidth(sideBarSize);
    view->setPosition(0,0);
    view->add("PixelStretcher");
    view->add("colorsdflk");
    view->add("sdfsdf");
    view->add("sdhf");
    view->add("hhhfh");
    view->add("dfhdfhs");
    view->setLabelAlignment(ofxDatGuiAlignment::CENTER);
    
    recordButton = new ofxDatGuiButton("record");

    recordButton->setWidth(sideBarSize);

    recordButton->setStripeColor(ofColor::red);

    recordButton->setBackgroundColor(ofColor(240,0,0));
    recordButton->setPosition(0,ofGetHeight()-28);
    recordButton->setLabelAlignment(ofxDatGuiAlignment::CENTER);

}

void GuiApp::update(){
    view->update();
    recordButton->update();

}

void GuiApp::draw(){
    ofSetColor(70);
    ofDrawRectangle(0, 0, sideBarSize, ofGetHeight());
    view->draw();
    recordButton->draw();

    mainApp->module->drawGui();

}
```
And some other gui code that is created in another window but drawn to the gui window

```
class Module_01: public PixelVerktModule
{
public:
    Module_01():PixelVerktModule()
    {
        

    }
    
    ///GUI
    
    ofxDatGui* gui1;
    ofxDatGui* gui2;
    ofxDatGui* gui3;

    void setupGui()
    {
        int columnWidth = 230;
        int sideBarSize = 110;
        gui1 = new ofxDatGui();
//        gui1->setTheme(new ofxDatGuiThemeWireframe());
        gui1->setWidth(columnWidth);
        gui1->setPosition(sideBarSize, 0);
        gui1->addSlider("sdfj", 0, 1);
        gui1->addSlider("sdfjh", 0, 1);
        gui1->addSlider("truc", 0, 1);
        gui1->addSlider("dd", 0, 1);
        gui1->addColorPicker("dff");
        gui1->setAutoDraw(false);
        
        gui2 = new ofxDatGui();
//        gui2->setTheme(new ThemeControls());
        gui2->setWidth(columnWidth);
        gui2->setPosition(sideBarSize+columnWidth, 0);
        gui2->addSlider("sdfj", 0, 1);
        gui2->addSlider("sdfjh", 0, 1);
        gui2->addSlider("truc", 0, 1);
        gui2->addSlider("dd", 0, 1);
        gui2->addColorPicker("dff");
        gui2->setAutoDraw(false);
        
        gui3 = new ofxDatGui();
//        gui3->setTheme(new ThemeControls());
        gui3->setWidth(columnWidth);
        gui3->setPosition(sideBarSize+columnWidth*2, 0);
        gui3->addColorPicker("dff");
        gui3->add2dPad("sdflk", ofRectangle(0, 0, 1, 1));
        gui3->addSlider("sdfj", 0, 1);
        gui3->addSlider("sdfjh", 0, 1);
        gui3->addSlider("truc", 0, 1);
        gui3->addSlider("dd", 0, 1);
        gui3->setAutoDraw(false);
    }
    
    void drawGui()
    {
        gui1->update();
        gui2->update();
        gui3->update();
        gui1->draw();
        gui2->draw();
        gui3->draw();
    }

};
```

**Comment by thomwolf** [comment]
I had a similar issue when I was setting GL version to 3+ on Mac OS X 10.8.
Going back to OpenGL 2 (default) fixed the problem.

**Comment by P-v-K** [comment]
Why is this issue closed? Did you find the cause or how to fix it, or just give up?

I am seeing the same issue and trying to figure out why it happens or how to fix it.

**Comment by P-v-K** [comment]
> 
> 
> I had a similar issue when I was setting GL version to 3+ on Mac OS X 10.8.
> Going back to OpenGL 2 (default) fixed the problem.

@thomwolf I don't know what you mean by setting GL version. Where? (I am seeing this problem intermittently on Android), just with ofxDatGui, not with ofTrueTypeFont.)

**Comment by P-v-K** [comment]
I found a fix for my situation (Android, random occurence, unsure of cause).

ofxSmartFont::add() assumes that if it finds a file with the same name and size as one being allocated, that it can just return a pointer to that.  But that's a bad assumption, for some reason I don't know. Best fix would be to otherwise confirm the font found actually has good font data. The hack I used instead just always allocates a new font on add().
